### PR TITLE
Samples Table: Fixes preserving page size when searching

### LINKED ISSUE #736: Samples Table: Preserve page size when searching
## Problem
When a user changes the number of samples to display per page and then initiates a search via search box or filter by sample puids/names, the page size resets to the default of 20 items.

## Acceptance criteria
* `limit` param is included when POSTing to samples search endpoint
* `limit` param if present is included in `redirect_to` within `search` action of samples controllers
* `page` param should not be preserved when searching

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Persisting the page size during sample filtering.
Fixes #736.

## Screenshots or screen recordings
image
image

## How to set up and validate locally
1. Navigate to a group or project samples page.
2. Select a page size other than the default.
3. Filter the samples.
4. Verify the page size remain unchanged.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/components/samples/table_component.html.erb [hunk 1]
```diff
@@ -220,7 +220,11 @@
   <% end %>
   <% if @has_samples %>
     <div class="flex items-center justify-between">
-      <%= render Viral::Pagy::LimitComponent.new(@pagy, item: t(".limit.item")) %>
+      <%= render Viral::Pagy::LimitComponent.new(
+        @pagy,
+        item: t(".limit.item"),
+        id: "samples_table_page_limit",
```
[the remark this answers is not in the document]
Comment (ksierks) [reply app/components/samples/table_component.html.erb]: I can. The concern is if there are more than one LimitComponent rendered on the page. That's why I added "samples_table" to the ID. Just trying to keep the IDs unique. Here you go, 1495c7324f5bbca1c3f85209d60daeebbf43f6a1!

**Review by ChrisHuynh333 — DISMISSED** [review]
Looks good!

File: app/components/viral/pagy/limit_component.html.erb [hunk 2]
```diff
@@ -1,11 +1,9 @@
-<div
-  class="inline-flex items-center mt-4 space-x-2  text-slate-500 dark:text-slate-400"
->
+<%= render Viral::BaseComponent.new(**system_arguments) do %>
```
Comment (joshsadam) [inline app/components/viral/pagy/limit_component.html.erb:1]: **nitpick:** Sorry can you just revert the changes to these two files and just add the id directly to the div.  I think it will be easier to read.  The div does not need any extra things passed into it.

Comment (ksierks) [reply app/components/viral/pagy/limit_component.html.erb]: 6a47e048399f7821455682eb0156e895003c9eaf

**Review by joshsadam — APPROVED** [review]
Looks great to me, thanks for the changes
